src/Databases/DatabaseOnDisk.cpp

```diff
@@ -490,14 +490,8 @@ ASTPtr DatabaseOnDisk::getCreateDatabaseQuery() const
     ASTPtr ast;
 
     auto settings = getContext()->getSettingsRef();
-    {
-        std::lock_guard lock(mutex);
```

Comment:
why lock is no longer needed? Not sure, but looks like it was needed to avoid race on metadata in `parseQueryFromMetadata`